### PR TITLE
[RPA-1703.2] Credential change email notification - email update

### PR DESCRIPTION
**Pull request by Aeri4a**
…cise edit operation info

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.utils.ts [hunk 1]
```diff
@@ -23,12 +24,14 @@ interface BaseCredentialChangeMailPayload {
     collectionCreatorEmail: string;
     collectionName: string;
     credentialName: string;
+    credentialOldName?: string | undefined;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.utils.ts]: If undefined is allowed, and this is the only use of the interface, why to make credentialOldName optional?

**Review by star-in-the-jar — APPROVED** [review]
+ version

File: runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.service.ts [hunk 2]
```diff
@@ -71,6 +71,14 @@ export class CredentialService {
...
 
     return this.credentialRepo.save(credential)
+      .then((credential) => {
+        this.notifyCredentialCollectionOwner({
+          executor: user,
+          collectionId: credential.collectionId,
+          credentialName: credential.name,
+          operationType: CredentialOperationType.CREATE,
+      });
+      })
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.service.ts]: can you lint this?

File: runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.utils.ts [hunk 3]
```diff
@@ -23,12 +24,14 @@ interface BaseCredentialChangeMailPayload {
...
 }
 
 interface BaseCredentialNotifyMailArgs {
     executor: IUser;
     collectionId: string;
     credentialName: string;
+    credentialOldName?: string | undefined;
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.utils.ts:34]: I think you should decide if you want to require this property, and if you do, passed value can be only string or undefined, or if you want to set this parameter as optional and always require a string if this property is defined

File: runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.service.ts [hunk 4]
```diff
@@ -222,13 +230,17 @@ export class CredentialService {
       tenantId: credential.tenantId,
     });
 
+    const credentialOldName =
+        credential.name !== credentialDto.name ? credential.name : undefined;
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.service.ts:235]: this should be adjusted depending on what approach you're gonna take in the interfaces

**Review by marcin-depcik — APPROVED** [review]

File: runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.utils.ts (hunk 1 above)
Comment (Aeri4a) [reply runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.utils.ts]: You're right

**Pull request merged**